keps/sig-security/2763-ambient-capabilities/README.md

```diff
@@ -0,0 +1,920 @@
+
+```
+FROM k8s.gcr.io/debian-base:buster-v1.4.0
+RUN apt-get update \
+    && apt-get -y --no-install-recommends install libcap2-bin
+COPY start-server /start-server
+RUN setcap cap_net_bind_service=+ep /start-server
+
+FROM gcr.io/distroless/static
+COPY --from=0 /start-server /usr/local/bin/start-server
+```
+
+**Now the following works:-**
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-web
+  labels:
+    role: myrole
+spec:
+  containers:
+    - name: web
+      image: ubuntu
+      command: ["start-server", "-port 80"]
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 1000
+        capabilities:
+          drop: ["All"]
+          add: ["NET_BIND_SERVICE"]
+      ports:
+      - name: web
+        containerPort: 80
+        protocol: TCP
+```
+
+**Note 1:** We cannot set `allowPrivilegeEscalation: false` anymore because `allowPrivilegeEscalation` directly controls the `no_new_privs` flag. With `no_new_privs` set, file capabilities are not added to the permitted set.
+
+**Note 2:** While applying the capabilities to the binary during the image build is a work-around, it should be noted that now even if we switched back to port 8080 in the above example we would have to add the capability in the container's `SecurityContext`. 
+
+While applying capabilities to the binary during image build is a work-around it is not always feasible to do so as a lot of users use 3rd party images, so in order to run containers as non-root in an image whose build you do not control you would essentially have to do something like:
+
+```Dockerfile
+FROM debian-base:buster-v1.4.0
+COPY --from=imageThatIWantToRunAsNonRoot /binaryThatIWantToRunAsNonRoot /binaryThatIWantToRunAsNonRoot
+RUN apt-get update \
+    && apt-get -y --no-install-recommends install libcap2-bin
+RUN setacp cap_my_binary_needs=+ep /binaryThatIWantToRunAsNonRoot
+
+FROM imageThatIWantToRunAsNonRoot
+# override the binary in the image
+COPY --from=0 /binaryThatIWantToRunAsNonRoot /binaryThatIWantToRunAsNonRoot
+```
+
+Here we override the binary in the original image with a setcaped binary and push the image to our private repository. While this is a workaround to the problem if you don’t control the build, you are now forced to effectively maintain another copy of the image.
+
+The way capabilities are today designed is that they are effective in limiting the capabilities of a user but we lack the ability to grant capabilities to a user unless we also control the build of an image. 
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Enable support for ambient capbilities for kubernetes and the `containerd` and `CRI-O` runtime.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Runtimes other than `containerd` and `CRI-O` are not in scope at the moment.
+
```

Comment:
For completeness, might be worth calling out that windows workloads are out of scope

Reply:
Done!